src/gate_kernels.jl

```diff
@@ -403,9 +407,23 @@ function apply_controlled_gate!(
     target_1::Int,
     target_2::Int,
 ) where {T<:Complex}
-    n_amps, (endian_control, endian_t1, endian_t2) =
-        get_amps_and_qubits(state_vec, control, target_1, target_2)
-    small_t, mid_t, big_t = sort([endian_control, endian_t1, endian_t2])
+    n_amps, (endian_control, endian_t1, endian_t2) = get_amps_and_qubits(state_vec, control, target_1, target_2)
+    small_t = min(endian_control, endian_t1, endian_t2)
+    big_t   = max(endian_control, endian_t1, endian_t2)
+    # this big if/else is to avoid an allocation
+    mid_t   = if     small_t == endian_control && big_t == endian_t1
+                  endian_t2
+              elseif small_t == endian_control && big_t == endian_t2
+                  endian_t1
+              elseif small_t == endian_t1 && big_t == endian_t2
+                  endian_control
+              elseif small_t == endian_t2 && big_t == endian_t1
+                  endian_control
+              elseif small_t == endian_t1 && big_t == endian_control
+                  endian_t2
+              elseif small_t == endian_t2 && big_t == endian_control
+                  endian_t1
+              end
```

Comment:
what about this, if you want to get overly clever:
```suggestion
    mid_t   = max(min(endian_control, endian_t1), min(endian_control, endian_t2), min(endian_t1, endian_t2))
```